File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/notice/exception/NoticeExceptionCode.java
package b1nd.dodam.domain.rds.notice.exception;

import b1nd.dodam.core.exception.ExceptionCode;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum NoticeExceptionCode implements ExceptionCode {

    NOT_APPLICANT(403, "작성자가 아님"),
    NOT_FOUND(404, "찾을 수 없음");

    private final int status;
    private final String message;

    @Override
    public int getStatus() {
        return this.status;
    }

    @Override
    public String getExceptionName() {
        return this.name();
    }

    @Override
    public String getMessage() {
        return this.message;
    }

}

File: services/service-notice/src/main/kotlin/com/b1nd/dodamdodam/notice/domain/notice/exception/NoticeExceptionCode.kt
package com.b1nd.dodamdodam.notice.domain.notice.exception

import com.b1nd.dodamdodam.core.common.exception.ExceptionCode
import org.springframework.http.HttpStatus

enum class NoticeExceptionCode(
    override val status: HttpStatus,
    override val message: String
): ExceptionCode {
    NOTICE_NOT_FOUND(HttpStatus.NOT_FOUND, "공지사항을 찾을 수 없어요."),
    NOT_NOTICE_AUTHOR(HttpStatus.FORBIDDEN, "공지사항 작성자가 아니에요."),
    INVALID_NOTICE_REQUEST(HttpStatus.BAD_REQUEST, "잘못된 공지사항 요청이에요."),
}
